src/hw/hw_mngr.cpp

```diff
@@ -357,3 +407,166 @@ void HWManager::setFanSpeedDelayed()
         setFanSpeed();
     });
 }
+
+void HWManager::detectFansDelayed()
+{
+    static boost::asio::deadline_timer tmr(io);
+
+    //  if we has expired attempts, then stop fan detection and work in normal
+    //  state
+    if (++numErrorAttempts > maxErrorAttempts)
+    {
+        log<level::ERR>("FAN Detection",
+                        entry("WHAT=reached maxErrorAttempts in state=%d",
+                              int(detectFansState)));
+        detectFansState = FanState::normal;
+        return;
+    }
+
+    // this implicitly cancels the timer
+    tmr.expires_from_now(boost::posix_time::seconds(5));
+    tmr.async_wait([&](const boost::system::error_code& err) {
+        if (err == boost::asio::error::operation_aborted)
+        {
+            /* we were canceled*/
+            return;
+        }
+        else if (err)
+        {
+            log<level::ERR>("Timer error",
+                            entry("WHAT=%s", err.message().c_str()));
+            return;
+        }
+
+        detectFansByRpmSpeed();
+    });
+}
+
+void HWManager::detectFansByRpmSpeed()
+{
+    log<level::DEBUG>("FAN: process detection by maximum RPM",
+                      entry("SYSTEM=current state=%d", int(detectFansState)));
```

Comment:
And here too. I guess something like `entyr("FANS_STATE=%d", int(detectFansState))` is more relevant here.
Moreover, current implementation looks invalid and I guess this won't work at all.
Unfortunately `phosphor-logging` keeps silence during compilation in case of wrong `entries` and then do nothing in runtime with the such wrong calls.

Reply:
Whole code and all log messages I always check during work on task. If somthing is not working I am trying to find a new solution. Do I right understand that "wrong entries" is mean "bad format string or incorrect argument types" ?  I didn't working with phosphor-logging sources, but... on other sources (in the past) I saw validation of format string to argument types and compilation was failed on programmer's errors.

FANS_STATE fixed.